lighthouse-core/audits/byte-efficiency/unminified-javascript.js

```diff
@@ -61,7 +61,7 @@ class UnminifiedJavaScript extends ByteEfficiencyAudit {
     const wastedBytes = Math.round(totalBytes * wastedRatio);
 
     return {
-      url: networkRecord.url,
+      url: networkRecord ? networkRecord.url : '?',
```

Comment:
> seems like we always know the undefined case is the main page

admittedly we don't know that will always be the case, so maybe it's better to make that certain in the gatherer

Reply:
> seems like we always know the undefined case is the main page

I think the problem here is inline scripts/evals added via other scripts. If it works like CSS those tend to be "inline" but shouldn't be considered part of the main page. I think the preference should be for code snippets showing the code itself, but I'd be more comfortable conveying it's in the "Other" / "Inline" / "Eval" bucket than the main page.

Reply:
Oh I gotcha, yeah agreed, we should just nix the finding of the main resource in the gatherer 👍 

I think just putting `inline` without giving any idea to the content is pretty rough UX. Maybe we can at least show the first 40 characters or something?

Reply:
I don't like omitting the `requestId` if it is readily available for inline content. How about not using an inline script's `requestId` in the audit, but keeping it in the artifact so that something else might use it if needed w/o grabbing the main record manually?

new pic: 
![image](https://user-images.githubusercontent.com/4071474/53520274-24465e00-3a8a-11e9-8f78-ce21f90ea113.png)